server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -2006,6 +2006,18 @@ public void testRelocatedShardCanNotBeRevived() throws IOException {
         closeShards(shard);
     }
 
+    public void testRecoveringShardFailsIfStartedTooSoon() throws IOException {
+        final IndexShard shard = newShard(randomBoolean());
```

Comment:
I think this must be:
```suggestion
        final IndexShard shard = newShard(false);
```

to not trigger the new assertion.

Reply:
yep that's why CI failed 🤦